docs/howto/krustlet-on-wsl2.md

```diff
@@ -0,0 +1,148 @@
+kind: CertificateSigningRequest
+metadata:
+  name: krustlet
+spec:
+  request: $(cat krustlet.csr | base64 | tr -d '\n')
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+EOF
+certificatesigningrequest.certificates.k8s.io/krustlet created
+```
+
+Once that runs, you will need to approve the request:
+
+```shell
+$ kubectl certificate approve krustlet
+certificatesigningrequest.certificates.k8s.io/krustlet approved
+```
+
+After approval, you can download the cert like so:
+
+```shell
+$ kubectl get csr krustlet -o jsonpath='{.status.certificate}' | base64 --decode > krustlet.crt
+```
+
+Lastly, combine the key and the cert into a PFX bundle, choosing your own password instead of
+"password":
+
+```shell
+$ openssl pkcs12 -export -out certificate.pfx -inkey krustlet.key -in krustlet.crt -password "pass:password"
+```
+
+## Step 2: Determine the default gateway
+
+The default gateway for most Docker containers is generally `172.17.0.1`.
+This IP is only reachable, by default, from the WSL2 distro.
+However, the `eth0` host network is accessible from Windows, so we can use this IP address to connect to the WSL2 distro (where Krustlet is running). 
+
+If this was changed, check `ifconfig eth0` from
+the host OS to determine the default gateway:
+
+```console
+$ ifconfig eth0
+eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
+        inet 172.26.47.208  netmask 255.255.240.0  broadcast 172.26.47.255
+        inet6 fe80::215:5dff:fe98:ce48  prefixlen 64  scopeid 0x20<link>
+        ether 00:15:5d:98:ce:48  txqueuelen 1000  (Ethernet)
+        RX packets 16818  bytes 11576089 (11.0 MiB)
+        RX errors 0  dropped 0  overruns 0  frame 0
+        TX packets 1093  bytes 115724 (113.0 KiB)
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+```
+
+In this example, I should use `172.26.47.208`.
+
+The hostname being "applied" from Windows, the default hostname will not resolve to this address, therefore you need to pass the `--node-ip` and `--node-name` flag to Krustlet.
+
+### Add a route to 172.17.0.1
+
+As stated above, the Docker default gateway, `172.17.0.1`, cannot be reached from Windows by default.
+
+However, a route can create to reach it, using the WSL2 own default gateway.
+
+We can use the following commands to create a temporary route:
+
+```shell
+PS> $env:WSLIP = Get-NetIPConfiguration -InterfaceAlias *WSL* | % { $_.IPv4Address.IPAddress }
+
+PS> $env:WSLIP = Get-NetIPConfiguration -InterfaceAlias *WSL* | % { $_.IPv4Address.IPAddress }
+ OK!
+```
+
+**DO NOT** make this route permanent as the WSL2 default gateway is DHCP based and will change upon every reboot.
+
+## Step 3: Install and run Krustlet
+
+First, install the latest release of Krustlet following [the install guide](../intro/install.md).
+
+Once you have done that, run the following commands to run Krustlet's WASI provider:
+
+```shell
+$ krustlet-wasi --node-ip 172.17.0.1 --node-name krustlet --pfx-password password
```

Comment:
instead of the gateway IP, use the actual IP address in wsl `172.26.47.208`.
Also the pfx password isn't required anymore if you use the context from above.
so it can be simplified to:
`KRUSTLET_NODE_IP=172.26.47.208 just run-wasi`

Reply:
thank you, I will remove the password option and will amend the `--node-ip`